src/com/seafile/seadroid2/transfer/TransferManager.java

```diff
@@ -83,6 +83,110 @@ public int addUploadTask(Account account, String repoID, String repoName,
         return task.getTaskID();
     }
 
+    private List<UploadTask> uploadWaitingList = Lists.newArrayList();
+    private List<UploadTask> uploadingList = Lists.newArrayList();
+    private static int mUploadingCount = 0;
+    private static final int UPLOAD_MAX_COUNT = 1;
+
+    /**
+     * add upload task to queue, execute one each time
+     * @param account
+     * @param repoID
+     * @param repoName
+     * @param dir
+     * @param filePath
+     * @param isUpdate
+     * @param isCopyToLocal
+     */
+    public void addTaskToUploadQue(Account account,
+                                   String repoID,
+                                   String repoName,
+                                   String dir,
+                                   String filePath,
+                                   boolean isUpdate,
+                                   boolean isCopyToLocal) {
+
+        UploadTask task = new UploadTask(account, repoID, repoName, dir, filePath, isUpdate, isCopyToLocal);
+        synchronized (uploadWaitingList) {
+            boolean hasInQue = false;
+            for (UploadTask d : uploadWaitingList) {
+                if (d.getTaskID() == task.getTaskID()) {
+                    hasInQue = true;
+                    break;
+                }
+            }
```

Comment:
Here the logic is confusing me. A newly created `UploadTask` (or `DownloadTask`) always have a new task ID, so here `hasInQue` would never because tru.

 To detect if there already exists  a task for the same file, we should iterate all the existing tasks and compare their fields.

What's more, I think  `addTaskToUploadQue` should call `addUploadTask`, and we should modify `addUploadTask` so it doesn't execute task immediately.